packages/flutter_tools/test/commands.shard/hermetic/run_test.dart

```diff
@@ -1045,19 +1042,19 @@ class FakeDevice extends Fake implements Device {
       }
       _throwToolExit(kSuccess);
     }
-    return null;
+    throw UnimplementedError();  // Unreachable.
```

Comment:
This shouldn't be needed if you update the return type of `_throwToolExit` on line 950 from `void` -> `Never`